internal/ingress/controller/handlers/node.go

```diff
@@ -67,11 +70,21 @@ func (h *EnqueueRequestsForNodeEvent) enqueueImpactedIngresses(queue workqueue.R
 		if !class.IsValidIngress(h.IngressClass, &ingress) {
 			continue
 		}
-		queue.Add(reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Namespace: ingress.Namespace,
-				Name:      ingress.Name,
-			},
-		})
+
+		namespacedName := types.NamespacedName{
+			Namespace: ingress.Namespace,
+			Name:      ingress.Name,
+		}
+		key := namespacedName.String()
+		ingressAnnos, err := h.Store.GetIngressAnnotations(key)
```

Comment:
the targetType annotation is supported on service as well. So you can even have different targetType in a single ingress.
We should check per service and only ignore node event if none of them are instance target.